source/particle/world.cc

```diff
@@ -1333,27 +1333,31 @@ namespace aspect
           sim->initialize_simulator (this->get_simulator());
         generator->parse_parameters(prm);
         generator->initialize();
+        generator->set_particle_world_index(world_index);
```

Comment:
This is the wrong order, the `parse_parameters` function will already need to know this index. Here and for the other functions.